lib/src/resolver.dart

```diff
@@ -1,87 +1,61 @@
-  static const PACKAGE_PREFIX = 'package:';
-  static const FILE_PREFIX = 'file://';
-
   final String packageRoot;
   final String sdkRoot;
-  final List<String> failed = [];
 
   Resolver({this.packageRoot, this.sdkRoot});
 
   /// Returns the absolute path wrt. to the given environment or null, if the
   /// import could not be resolved.
-  String resolve(String uri) {
-    if (uri.startsWith(DART_PREFIX)) {
+  String resolve(Uri uri) {
+    if (uri.scheme == 'dart') {
       if (sdkRoot == null) {
         // No sdk-root given, do not resolve dart: URIs.
         return null;
       }
-      var slashPos = uri.indexOf('/');
-      var filePath;
-      if (slashPos != -1) {
-        var path = uri.substring(DART_PREFIX.length, slashPos);
-        // Drop patch files, since we don't have their source in the compiled
-        // SDK.
-        if (path.endsWith('-patch')) {
-          failed.add(uri);
-          return null;
-        }
-        // Canonicalize path. For instance: _collection-dev => _collection_dev.
-        path = path.replaceAll('-', '_');
-        filePath = '$sdkRoot/${path}${uri.substring(slashPos, uri.length)}';
+
+      // Resolve 'dart:something' to be something/something.dart in the SDK.
+      var lib = uri.pathSegments.first;
+
+      var pathSegments = [sdkRoot, 'lib', lib];
+
+      if (uri.pathSegments.length > 1) {
+        pathSegments.addAll(uri.pathSegments.skip(1));
       } else {
-        // Resolve 'dart:something' to be something/something.dart in the SDK.
-        var lib = uri.substring(DART_PREFIX.length, uri.length);
-        filePath = '$sdkRoot/$lib/${lib}.dart';
+        pathSegments.add('${lib}.dart');
       }
-      return resolveSymbolicLinks(filePath);
+
+      var filePath = p.joinAll(pathSegments);
+
+      return _resolveSymbolicLinks(filePath);
     }
-    if (uri.startsWith(PACKAGE_PREFIX)) {
+    if (uri.scheme == 'package') {
       if (packageRoot == null) {
         // No package-root given, do not resolve package: URIs.
         return null;
       }
-      var path = uri.substring(PACKAGE_PREFIX.length, uri.length);
-      return resolveSymbolicLinks(p.join(packageRoot, path));
+      return _resolveSymbolicLinks(p.join(packageRoot, uri.path));
     }
-    if (uri.startsWith(FILE_PREFIX)) {
-      return resolveSymbolicLinks(p.fromUri(Uri.parse(uri)));
+    if (uri.scheme == 'file') {
+      return _resolveSymbolicLinks(p.fromUri(uri));
     }
-    // We cannot deal with anything else.
-    failed.add(uri);
-    return null;
-  }
 
-  /// Returns a canonicalized path, or `null` if the path cannot be resolved.
-  String resolveSymbolicLinks(String path) {
-    var normalizedPath = p.normalize(path);
-    var type = FileSystemEntity.typeSync(normalizedPath, followLinks: true);
-    if (type == FileSystemEntityType.NOT_FOUND) return null;
-    return new File(normalizedPath).resolveSymbolicLinksSync();
+    throw new Exception('Could not resolve "$uri"');
```

Comment:
@kevmoo 
i'm execute dartium and collect coverage info. The previous version(7.3) of collecting coverage information without error, because it can handle the url. Example http://localhost:8080/my_browser_test.dart and etc.
Reproduce steps:
1) `pub serve test`
2) `pub run test --concurrency=1 --pub-serve=8080 -p dartium`
3) wait of the end of the tests. I used special test package, it is not kill isolations and dartium. So I can collect all the data coverage.
4) `pub global run coverage collect --port=<REMOTE_PORT> --host=localhost -o coverage.lcov`
 Data collection falls because the method `String resolve(Uri uri) [lib/src/resolver.dart]` comes url = `'http://localhost:8080/my_browser_test.dart'`